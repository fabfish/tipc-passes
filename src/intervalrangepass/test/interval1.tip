main() {
  var x, y, z;
  x = 42;	// x = [42,42]
  if (input) 
    z = x;	// z = [42,42]
  else {
    z = 4;	// z = [4,4]
  }
  return z;	// z = [4,42]
}

