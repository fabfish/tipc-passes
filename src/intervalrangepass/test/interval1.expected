*** interval range analysis for function define i64 @_tip_main() {
entry:
  %0 = call i64 @_tip_input()
  %ifcond = icmp ne i64 %0, 0
  br i1 %ifcond, label %then1, label %else1

then1:                                            ; preds = %entry
  br label %ifmerge1

else1:                                            ; preds = %entry
  br label %ifmerge1

ifmerge1:                                         ; preds = %else1, %then1
  %z.0 = phi i64 [ 42, %then1 ], [ 4, %else1 ]
  call void @llvm.donothing()
  ret i64 %z.0
}
 ***
  %ifcond = icmp ne i64 %0, 0 = [0,1]
  %0 = call i64 @_tip_input() = [-inf,+inf]
  %z.0 = phi i64 [ 42, %then1 ], [ 4, %else1 ] = [4,42]
  call void @llvm.donothing() = [-inf,+inf]
